core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/scheduling/ScheduleManagerImpl.java

```diff
@@ -151,11 +167,16 @@ private void tryRunNext()
         {
             for (ScheduledJob next : myQueue)
             {
-                if (validate(next) && tryRunTasks(next))
+                if (validate(next))
                 {
-                    break;
+                    currentExecutingJob.set(next); // Set the currently executing job
+                    if (tryRunTasks(next))
+                    {
+                        break;
+                    }
                 }
             }
+            currentExecutingJob.set(null); // Reset after trying to run a job
```

Comment:
Same as above; why it is reset is better to say than stating the fact (if comment is necessary)